### ISSUE TITLE
[12.0.rc0 Version] qpdf.dll runtime not supported

### ISSUE THREAD

**Issue by Mr-MooMoo** [QuestPDF/QuestPDF#1057]
**Describe the bug**
I am currently testing the new Zugferd/PDF-Attachment features, but currently running in some problems with qpdf.dll (See Stack trace below).
qpdf.dll is copied to output directory, project set to win-x64 (see screenshot)
![image](https://github.com/user-attachments/assets/2102a082-d7a8-4826-9f67-caf53c846c47)
But quest-pdf seems to be upset anyway ;)
Maybe i am just missing something !

**To Reproduce**
[Here is a minimal WinFormsApp](https://github.com/Mr-MooMoo/Minimal-QuestPDF)

**Expected behavior**
As set to x64, it should work ;)

**Environment**
win-x64 on windows 11;  project set to net8 (net9 installed)

**Additional context**
Stack-Trace:
```
System.Exception: The QuestPDF library has encountered an issue while loading one of its dependencies.

Your runtime is currently not supported by QuestPDF. Currently supported runtimes are: win-x86, win-x64, linux-x64, linux-arm64, linux-musl-x64, osx-x64, osx-arm64. 

It appears that your current operating system distribution may be outdated. For optimal compatibility, please consider updating it to a more recent version.

Please always set the 'Platform target' to either 'X86' or 'X64' in your startup project settings. Please do not use the 'Any CPU' option.
 ---> System.DllNotFoundException: Unable to load DLL 'qpdf' or one of its dependencies: The specified module could not be found. (0x8007007E)
   at QuestPDF.Qpdf.QpdfAPI.API.qpdf_get_qpdf_version()
   at QuestPDF.Qpdf.QpdfAPI.GetQpdfVersion()
   at QuestPDF.Qpdf.QpdfNativeDependencyCompatibilityChecker.ExecuteNativeCode()
   at QuestPDF.Helpers.NativeDependencyCompatibilityChecker.CheckIfExceptionIsThrownWhenLoadingNativeDependencies()
   --- End of inner exception stack trace ---
   at QuestPDF.Helpers.NativeDependencyCompatibilityChecker.<TestOnce>g__ThrowCompatibilityException|13_0(Exception innerException)
   at QuestPDF.Helpers.NativeDependencyCompatibilityChecker.TestOnce()
   at QuestPDF.Helpers.NativeDependencyCompatibilityChecker.Test()
   at QuestPDF.Qpdf.QpdfNativeDependencyCompatibilityChecker.Test()
   at QuestPDF.Qpdf.QpdfAPI.ExecuteJob(String jobJson)
   at QuestPDF.Fluent.DocumentOperation.Save(String filePath)
   at M.......WebUI.PDFEngine.PDFEngine.GenerateInvoiceWithZugferdData(OrderDto orderDto) in S:\repos\.......WebUI\PDFEngine\PDFEngine.cs:line 45
   at M.......WebUI.Components.Orders.OrderListActionMenu.DebugZugferd(OrderDto order) in S:\repos\M.......\Components\Orders\OrderListActionMenu.razor:line 309
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at MudBlazor.MudMenuItem.OnClickHandlerAsync(MouseEventArgs ev)
   at MudBlazor.MudListItem`1.OnClickHandlerAsync(MouseEventArgs eventArgs)
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
```

**Comment by MarcinZiabek** [comment]
Thank you for sharing this issue! 😄 

I have noticed that you are using Blazor. I am not very familiar with this library. Could you please check if the QuestPDF logic is executed in the c# backend code, or as part of JavaScript / frontend / WebAssembly?

**Comment by Mr-MooMoo** [comment]
Hey there (and thank you for your amazing work by the way ;)!   
Indeed in the stack trace provided i was using blazor - but the server version, so everything was executed backend-wise.
But to focus on the problem, i started a simple winforms-App (see my link to my repo above), and I got the same error there.

**Comment by KronerPa** [comment]
I got some information related to this issue to share. On my private desktop the qpdf library could be used to add an attachment even with AnyCPU setting.
It's a bit hard to tackle down the origin of the problems, but i try to share as much information as i can.

Working device:
- Win11 Education | 23H2 | 22631.4460 | Upgraded from Win10

NotWorking device:
- Win11 Home | 24H2 | 26100.2314 | No upgrade from Win10 | core isolation is active

**Comment by KronerPa** [comment]
It seems that there is something of with copying the necessary files into the folders. I ran dependency walker through both machines and detected the following:

![CleanDependencies](https://github.com/user-attachments/assets/677d8687-f8b5-4bac-b575-fe31d41fec44)

Additionally the qpdf and QuestPdfSkia dll is copied into the executable directory instead of the runtime directories. The missing dlls arent found anywhere.
![image](https://github.com/user-attachments/assets/158011c3-2b87-4fc0-ab26-a8c9d679d37e)

**Comment by KronerPa** [comment]
The issue is resolved on my devices with using 2024.12.0-rc2 @Mr-MooMoo @MarcinZiabek

**Comment by Mr-MooMoo** [comment]
Nice - seems to work here, too!   I have too little time to investigate in detail right now, but at least I get some totally different errors now ;)
